frame/contracts-registry/src/types.rs

```diff
@@ -0,0 +1,15 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{sp_runtime::RuntimeDebug, traits::Get, BoundedVec};
+use scale_info::TypeInfo;
+
+pub type ContractIndex = u32;
+
+#[derive(Clone, Encode, Decode, MaxEncodedLen, RuntimeDebug, TypeInfo)]
+#[scale_info(skip_type_params(MaxContractsLen, MaxParties))]
+pub struct ContractDataStorage<MaxContractsLen: Get<u32>, MaxParties: Get<u32>, AccountId> {
+	// Content of contract
+	pub data: BoundedVec<u8, MaxContractsLen>,
+	// Vec of parties that may sign following contract
+	pub parties: BoundedVec<AccountId, MaxParties>,
+	pub creator: AccountId,
```

Comment:
We don't care about the creator in any on-chain logic, so lets not waste blockchain-space on it. We can always index it offchain

Reply:
From what I remember the requirement is that anyone can delete a contract as long as there are no signatures. We don't need to know creator for that.

Reply:
But then anyone can snipe deposit. I think it is much clear to leave remove only for creator or  judges